### ISSUE TITLE
Compilation failure with gcc-11

### ISSUE THREAD

**Issue by bjacklyn** [ARM-software/gator#28]
```
In file included from linux/perf/PerfBuffer.cpp:9:
./linux/perf/PerfBuffer.h:27:9: error: 'size_t' does not name a type
   27 |         size_t pageSize;
      |         ^~~~~~
./linux/perf/PerfBuffer.h:18:1: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
   17 | #include "Config.h"
  +++ |+#include <cstddef>
```

**Comment by bengaineyarm** [comment]
Hi @bjacklyn - thanks for this. 

What tag / version are you attempting to build/use.  I am not able to repro this for 7.5, 7.6, 7.7 or our current internal development build.

If you are using an older version, is their any reason you need the combination of old Streamline version + brand new GCC version?

**Comment by bjacklyn** [comment]
Ah I should have said the version -- it's `7.1` -- the reason for that is it's another team which decides which version of dependencies they want to use, whereas I'm just upgrading the compiler to the latest.

In the latest versions this must be transitively available from somewhere else, but I do think it's worthwhile to IWYU and it's also a trivial fix -- I've manged to workaround it by creating this patchfile in our build system:
```
diff --git a/daemon/linux/perf/PerfBuffer.h b/daemon/linux/perf/PerfBuffer.h
index 083ca4d..fdd0aae 100644
--- a/daemon/linux/perf/PerfBuffer.h
+++ b/daemon/linux/perf/PerfBuffer.h
@@ -9,6 +9,7 @@
 #ifndef PERF_BUFFER
 #define PERF_BUFFER
 
+#include <cstddef>
 #include <map>
 #include <set>
 #include <vector>
```

**Comment by bengaineyarm** [comment]
> Ah I should have said the version -- it's 7.1 -- the reason for that is it's another team which decides which version of dependencies they want to use, whereas I'm just upgrading the compiler to the latest.

Ok, thanks.

> In the latest versions this must be transitively available from somewhere else, but I do think it's worthwhile to IWYU and it's also a trivial fix

Yes agreed, thank for pointing it out.

> I've manged to workaround it by creating this patchfile in our build system:

Great

**Comment by clementperon** [comment]
https://github.com/ARM-software/gator/pull/40

**Comment by bengaineyarm** [comment]
Resolved by #40
